src/gui/search/indexersdialog.h

```diff
@@ -26,39 +27,49 @@
  * exception statement from your version.
  */
 
-#include "pluginsourcedialog.h"
+#pragma once
 
-#include "gui/utils.h"
-#include "ui_pluginsourcedialog.h"
+#include <QDialog>
+#include <QStringList>
 
-#define SETTINGS_KEY(name) "SearchPluginSourceDialog/" name
+#include "base/search/searchengine.h"
 
-PluginSourceDialog::PluginSourceDialog(QWidget *parent)
-    : QDialog(parent)
-    , m_ui(new Ui::PluginSourceDialog)
-    , m_storeDialogSize(SETTINGS_KEY("Size"))
-{
-    m_ui->setupUi(this);
-    setAttribute(Qt::WA_DeleteOnClose);
-
-    Utils::Gui::resize(this, m_storeDialogSize);
-    show();
-}
+class QTreeWidgetItem;
 
-PluginSourceDialog::~PluginSourceDialog()
+namespace Ui
 {
-    m_storeDialogSize = size();
-    delete m_ui;
+    class IndexersDialog;
 }
 
-void PluginSourceDialog::on_localButton_clicked()
+class IndexersDialog final : public QDialog
 {
-    emit askForLocalFile();
-    close();
-}
+    Q_OBJECT
+    Q_DISABLE_COPY_MOVE(IndexersDialog)
 
-void PluginSourceDialog::on_urlButton_clicked()
-{
-    emit askForUrl();
-    close();
-}
+public:
+    explicit IndexersDialog(SearchEngine *searchEngine, QWidget *parent = nullptr);
+    ~IndexersDialog() override;
+
+private slots:
+    void on_actionEnable_toggled(bool enabled);
+    void on_actionEdit_triggered();
+    void on_actionRemove_triggered();
+    void on_addButton_clicked();
+    void on_closeButton_clicked();
+
+    void toggleIndexerState(QTreeWidgetItem *, int);
+    void displayContextMenu(const QPoint &);
+    void onIndexerAdded(const QString &indexerName);
+    void onIndexerOptionsChanged(const QString &indexerName);
+    void onIndexerStateChanged(const QString &indexerName);
+    void onIndexerRemoved(const QString &indexerName);
+
+private:
+    void loadIndexers();
+    void createItem(const QString &indexerName, const IndexerInfo &indexerInfo);
+    QTreeWidgetItem *findItemByName(const QString &indexerName);
```

Comment:
I'm astounded that almost none of them are `const` function, surely this one can be it, right?